module/builder/consensus/builder_test.go

```diff
@@ -730,3 +730,86 @@ func (bs *BuilderSuite) TestPayloadReceiptLimit() {
 	bs.Require().NoError(err)
 	bs.Assert().Equal(bs.pendingReceipts[:limit], bs.assembled.Receipts, "should have excluded receipts above maxReceiptCount")
 }
+
+// TestPayloadReceiptNoParentResult tests that the builder does not include
+// receipts whose PreviousResult is not already incorporated in the chain.
+//
+// Here we create 4 consecutive blocks S, A, B, and C, where A contains a valid
+// receipt for block S, but blocks B and C have empty payloads.
+//
+// We populate the mempool with valid receipts for blocks A, and C, but NOT for
+// block B.
+//
+// The expected behaviour, is that the builder should not include the receipt
+// for block C, because the chain and the mempool do not contain a valid receipt
+// for the parent result (block B's result).
+//
+// ... <- S <- A[ER{S}] <- B <- C <- X (candidate)
```

Comment:
FYI. I made a function that can [prepare such chain much easier](https://github.com/onflow/flow-go/pull/320/files#diff-77f09607e896b942749ffed547ac64d32d9007b8b564b86bd0c942dd7d063b0bR262-R272)

Reply:
Nice. Perhaps it makes sense that I target this PR at your branch: leo/validate-multi-receipts
Or I can wait until yours is merged

Reply:
yup, my PR has been merged to master